### ISSUE TITLE
Units of time on tooltip window

### ISSUE THREAD

**Issue by BasilioPV** [OpenModelica/OpenModelica#9950]
### Description

"Prefix Units" on plot window does not works: the value changes but the tooltip does not show the new units.

### Steps to Reproduce

Use any plot.

### Expected Behavior

Units on tooltip shall be according the quantity shown.

### Screenshots

![image](https://user-images.githubusercontent.com/93596576/207915907-cc5f2cfe-85b7-41fc-8e69-addea79f8fbc.png)

With "Prefix Units" activated:

![image](https://user-images.githubusercontent.com/93596576/207916014-f628bb6b-56f4-41e1-b317-17cecc76c890.png)

With "Prefix Units" deactivated (correct behaviour):

![image](https://user-images.githubusercontent.com/93596576/207916180-3b468eb5-3f8d-4cfd-84a0-482a154c8cc5.png)

### Version and OS

 - OpenModelica Version: 1.20.0
 - OS:  Windows 10, 64 bit
 - Modelica 4.0.0

### Additional Context

**Comment by casella** [comment]
@BasilioPV I'm not sure I understand what the problem is. I tried with this simple MWE:
```modelica
model M
  Real x(unit="m") = 1e5*sin(time*10);
end M;
```
This is with prefix units:
![immagine](https://user-images.githubusercontent.com/4233148/207944030-4939a742-1bd8-47e7-b238-594c68137c72.png)
and this is without prefix units
![immagine](https://user-images.githubusercontent.com/4233148/207944217-69522a09-e854-48b3-952c-ec2a975f09b0.png)
Looks good to me. Do I miss something?

The only problem I see is that when prefix units is changes in the plot setup, the axes are immediately updated, while the legend (including the unit) is only updated if you bring some other window to the foreground, and then bring OMEdit back to foreground, so that everything is redrawn from scratch.

@adeas31, you may want to fix this, so that also the legends are redrawn when prefix units is toggled.

**Comment by BasilioPV** [comment]
@casella the problem appears with time >100 sec. In these cases the time on the tooltip is divided by 100 but the units remain on "s"

**Comment by casella** [comment]
Aha, OK. Thanks for reporting!

@adeas31 please check the issue with this MWE:
```modelica
model M
  Real x(unit="m") = 1e5*sin(time/10);
annotation(StopTime = 1000);
end M;
```
![immagine](https://user-images.githubusercontent.com/4233148/208003628-e023d0dd-15ef-41fc-a75b-82627dcabaf9.png)

@adeas31 this should be fixed ASAP and ported to maintenance/v1.20. Thanks!

**Comment by bilderbuchi** [comment]
Note: This does not only affect the time axis, but also the y axis. With this MWE
```modelica
model PlotCursorScaling
  Modelica.Blocks.Sources.Sine sine(amplitude = 0.14/2, f = 2, offset = 0.14/2)  annotation(
    Placement(visible = true, transformation(origin = {-30, 16}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  Modelica.Units.SI.Length out = sine.y;
equation

annotation(
    uses(Modelica(version = "4.0.0")));
end PlotCursorScaling;
```
the plot cursor shows the correct values when prefix units the plot setup is false.
When prefix units is checked, even after switching windows, the values for `out` in the tooltip are incorrect (by a factor of 10 in this case).
![image](https://user-images.githubusercontent.com/327442/214528615-07c3333a-43fe-4f44-8407-9620540c7ce5.png)

**Comment by adeas31** [comment]
> Aha, OK. Thanks for reporting!
> 
> @adeas31 please check the issue with this MWE:
> 
> ```modelica
> model M
>   Real x(unit="m") = 1e5*sin(time/10);
> annotation(StopTime = 1000);
> end M;
> ```
> 
> ![immagine](https://user-images.githubusercontent.com/4233148/208003628-e023d0dd-15ef-41fc-a75b-82627dcabaf9.png)
> 
> @adeas31 this should be fixed ASAP and ported to maintenance/v1.20. Thanks!

One idea is to not show the time unit on tooltip same like value. What do you think?

**Comment by casella** [comment]
> One idea is to not show the time unit on tooltip same like value. What do you think?

I'm not sure I understand what you mean exactly. IMO if we use prefixes for the plotted quantity, we should also do it for time.

**Comment by bilderbuchi** [comment]
@adeas31 as the commit message only mentions "time", does this fix also fix the equivalent problem with y axis quantities that I reported [above](https://github.com/OpenModelica/OpenModelica/issues/9950#issuecomment-1403329718)?

**Comment by adeas31** [comment]
Yes, that's correct. What you found was a case where exponent was set to -1.

**Comment by casella** [comment]
@adeas31 I'm not sure if we are ever releasing 1.20.1, but in case 1.21.0 is delayed for any reason, this is a fix that should definitely provided in 1.20.1. Would you mind cherry picking your fixes in 1.20 maintenance as well?

Thanks!

**Comment by adeas31** [comment]
Done. See #10125
